seed/energy/meter_data_processor/green_button_parser.py

```diff
@@ -0,0 +1,170 @@
+import logging
+
+from seed.energy.meter_data_processor import green_button_maps
+_log = logging.getLogger(__name__)
+
+root_ns = '{http://www.w3.org/2005/Atom}'
+gb_ns = '{http://naesb.org/espi}'
+
+def get_link_href_head(link):
+    href = link.get('href')
+    if href[0:1]=='/':
+        href = href[1:]
+    first_slash = href.find('/')
+    return href[0:first_slash]
+
+def get_link_href_tail(link):
+    href = link.get('href')
+    last_slash = href.rfind('/')
+    return href[(last_slash+1):]
```

Comment:
Hi Piper,

I checked the urlparse library and found there is no help... Actually the
link is not a URL, it's a path, e.g. /espi/1_1/resource/User/01/UsagePoint.
After urlparse, the result only has path attribute which is the original
string...

Best,
Haopeng

On Thu, Feb 11, 2016 at 11:55 AM, Piper Merriam <notifications@github.com>
wrote:

> In seed/energy/meter_data_processor/green_button_parser.py
> <https://github.com/SEED-platform/seed/pull/714#discussion_r52630563>:
>
> > +_log = logging.getLogger(__name__)
> > +
> > +root_ns = '{http://www.w3.org/2005/Atom}'
> > +gb_ns = '{http://naesb.org/espi}'
> > +
> > +def get_link_href_head(link):
> > +    href = link.get('href')
> > +    if href[0:1]=='/':
> > +        href = href[1:]
> > +    first_slash = href.find('/')
> > +    return href[0:first_slash]
> > +
> > +def get_link_href_tail(link):
> > +    href = link.get('href')
> > +    last_slash = href.rfind('/')
> > +    return href[(last_slash+1):]
>
> You might consider using the urlparse
> <https://docs.python.org/2/library/urlparse.html> library for this
> functionality.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SEED-platform/seed/pull/714/files#r52630563>.
>